sdks/java/core/src/main/java/org/apache/beam/sdk/values/PCollection.java

```diff
@@ -301,6 +301,18 @@ public String getName() {
     return setCoder(SchemaCoder.of(schema, toRowFunction, fromRowFunction));
   }
 
+  /** Returns whether this {@link PCollection} has an attached schema. */
+  @Experimental(Kind.SCHEMAS)
+  public boolean hasSchema() {
+    return getCoder() instanceof SchemaCoder;
+  }
+
+  /** Returns the attached schema, or null if there is none. */
```

Comment:
I'm new to Java, but I believe throwing an exception (possibly `IllegalStateException`) might be more idiomatic here. If not that, this function needs a `@Nullable` annotation.

Reply:
done